src/test/java/gr/cytech/chatreminderbot/rest/ControlIT.java

```diff
@@ -163,6 +164,85 @@ void checkRemindFormatTest() {
                 "\" ,  \"thread\": { \"name\": \"spaces/" + "SPACE_ID" + "\" }}");
     }
 
+    @Test
+    void setMyTimezone() {
+        Request req = new Request();
+        Message mes = new Message();
+        Sender sender = new Sender();
+        ThreadM threadM = new ThreadM();
+
+
+        threadM.setName("space/SPACE_ID/thread/THREAD_ID");
+        sender.setName("MyName");
+
+        mes.setThread(threadM);
+        mes.setSender(sender);
+
+        mes.setText("@reminder set my timezone to athens");
+        req.setMessage(mes);
+
+        String expectedResponse = " <"+req.getMessage().getSender().getName()+"> successfully set your timezone at:Europe/Athens";
+
+        Client c = ClientBuilder.newBuilder().register(new JacksonJsonProvider(new ObjectMapper())).build();
+        Response resp = c.target("http://localhost:8080/bot/services/handleReq")
+                .request()
+                .post(Entity.json(req));
+        resp.bufferEntity();
+
+        assertThat(resp.readEntity(String.class)).isEqualTo("{ \"text\": \"" + expectedResponse +
+                "\" ,  \"thread\": { \"name\": \"spaces/" + "SPACE_ID" + "\" }}");
+    }
+
+
+
+
+    @Test
+    void setAndReturnTimezone() {
+        Request req = new Request();
+        Request req2 = new Request();
+        Message mes = new Message();
+        Message mes2 = new Message();
+        Sender sender = new Sender();
+        ThreadM threadM = new ThreadM();
+
+
+        threadM.setName("space/SPACE_ID/thread/THREAD_ID");
+        sender.setName("MyName");
+
+        mes.setThread(threadM);
+        mes.setSender(sender);
+
+        mes2.setThread(threadM);
+        mes2.setSender(sender);
+
+
+        mes2.setText("@reminder set my timezone to athens");
+        mes.setText("@reminder timezones");
+        req.setMessage(mes);
+        req2.setMessage(mes2);
```

Comment:
It's hard to track what's happening with all these beans being created and set.
Please rearrange the code, so that each request is created just before it is used as a request parameter.